valhalla/userrequests/state_changes.py

```diff
@@ -233,21 +233,22 @@ def update_request_states_for_window_expiration():
 
 def update_request_states_from_pond_blocks(pond_blocks):
     '''Update the states of requests and user_requests given a set of recently changed pond blocks.'''
-    sorted_blocks = sorted(pond_blocks, key=lambda x: x['molecules'][0]['tracking_number'])
-    blocks_by_tracking_num = itertools.groupby(sorted_blocks, lambda x: x['molecules'][0]['tracking_number'])
+    sorted_blocks = sorted(pond_blocks, key=lambda x: x['molecules'][0]['tracking_num'])
+    blocks_by_tracking_num = itertools.groupby(sorted_blocks, lambda x: x['molecules'][0]['tracking_num'])
     now = timezone.now()
     states_changed = False
 
     for tracking_num, blocks in blocks_by_tracking_num:
-        sorted_blocks_by_request = sorted(blocks, key=lambda x: x['molecules'][0]['request_number'])
-        blocks_by_request_num = {k: list(v) for k, v in itertools.groupby(sorted_blocks_by_request, key=lambda x: x['molecules'][0]['request_number'])}
-        user_request = UserRequest.objects.prefetch_related('requests').get(pk=tracking_num)
-        ur_expired = user_request.max_window_time < now
-        requests = user_request.requests.all()
-        for request in requests:
-            if request.id in blocks_by_request_num:
-                states_changed |= update_request_state(request, blocks_by_request_num[request.id], ur_expired)
-        states_changed |= update_user_request_state(user_request)
+        if tracking_num:
```

Comment:
What does this mean? Why would we have a variable `blocks_by_tracking_num` if there was no tracking number? Better to filter out the blocks that don't have a tracking number at line 237 in the lambda so they never make it in to the variable that is named to have tracking numbers.